File: tests/XMP/global-view/task/C-Round/tasks.c
#include <stdio.h>

#pragma xmp nodes p(8)

int main(){

#pragma xmp tasks
  {

#pragma xmp task on p(1:4)
    {
#pragma xmp barrier on p
    }

#pragma xmp task on p(5:8)
    {
#pragma xmp barrier on p
    }

  }

#pragma xmp task on p(1)
  {
    printf("PASS\n");
  }

  return 0;

}

File: tests/XMP/global-view/task/F/tasks.f90
program tasks

!$xmp nodes p(8)

!$xmp tasks

!$xmp task on p(1:4)
!$xmp barrier on p
!$xmp end task

!$xmp task on p(5:8)
!$xmp barrier on p
!$xmp end task

!$xmp end tasks

!$xmp task on p(1)
  write(*,*) "PASS"
!$xmp end task

end program tasks
